apps/chat/src/components/Chat/ModelList.tsx

```diff
@@ -185,6 +306,64 @@ export const ModelList = ({
   disabled,
   isReplayAsIs,
 }: ModelListProps) => {
+  const dispatch = useAppDispatch();
+
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+  const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+  const [currentEntity, setCurrentEntity] = useState<DialAIEntityModel>();
+  const [isPublishing, setIsPublishing] = useState(false);
+  const recentModelsIds = useAppSelector(ModelsSelectors.selectRecentModelsIds);
+  const applicationDetail = useAppSelector(
+    ApplicationSelectors.selectApplicationDetail,
+  );
+
+  const entityForPublish = currentEntity
+    ? ({
+        name: currentEntity.name,
+        id: ApiUtils.decodeApiUrl(currentEntity.id),
+        folderId: getFolderIdFromEntityId(currentEntity.id),
+      } as ShareEntity)
+    : null;
+
+  const handleEdit = (currentEntityId: string) => {
```

Comment:
as far as we drill these functions as a prop to child components, lets wrap it with useCallback